cf-oscore/src/main/java/org/eclipse/californium/oscore/Encryptor.java

```diff
@@ -160,17 +160,30 @@ protected static Encrypt0Message prepareCOSEStructure(byte[] confidential) {
 		int firstByte = 0x00;
 		ByteArrayOutputStream bRes = new ByteArrayOutputStream();
 		byte[] partialIV = OSSerializer.processPartialIV(ctx.getSenderSeq());
-		firstByte = firstByte | (partialIV.length & 0x07);
-		firstByte = firstByte | 0x08;
+		firstByte = firstByte | (partialIV.length & 0x07); //PartialIV length
+		firstByte = firstByte | 0x08; //Set the KID bit
+
+		//If the Context ID should be included for this context, set its bit
+		if(ctx.getIncludeContextId() == true) {
```

Comment:
Please insert a space (' ') between "if" and "(".

Reply:
And only keep the "== true" if you really prefer it :-).

Reply:
Fixed in commit c2bffa2b0b1144b65108adbd7f1cc08638928d44